sov-modules/sov-state/src/jmt_storage.rs

```diff
@@ -1,24 +1,50 @@
+use std::{cell::RefCell, rc::Rc};
+
 use crate::storage::{Storage, StorageKey, StorageValue};
-use first_read_last_write_cache::cache::CacheLog;
+use first_read_last_write_cache::{
+    cache::{self, CacheLog},
+    CacheValue,
+};
 use jellyfish_merkle_generic::Version;
 
 // Storage backed by JMT.
+#[derive(Default, Clone)]
 pub struct JmtStorage {
     // Caches first read and last write for a particular key.
-    _cache: CacheLog,
+    cache: Rc<RefCell<CacheLog>>,
     _version: Version,
 }
 
 impl Storage for JmtStorage {
-    fn get(&self, _key: StorageKey) -> Option<StorageValue> {
-        todo!()
+    fn get(&self, key: StorageKey) -> Option<StorageValue> {
+        let cache_key = key.into();
+        let cache_value = self.cache.borrow().get_value(&cache_key);
+
+        match cache_value {
+            cache::ExistsInCache::Yes(cache_value_exists) => {
+                self.cache
+                    .borrow_mut()
+                    .add_read(cache_key, cache_value_exists.clone())
+                    // It is ok to panic here, we must guarantee that the cache is consistent.
+                    .unwrap_or_else(|e| panic!("Inconsistent read from the cache: {e:?}"));
+
+                cache_value_exists.value.map(|value| StorageValue { value })
+            }
+            // TODO If the value does not exist in the cache, then fetch it from the JMT.
+            cache::ExistsInCache::No => todo!(),
+        }
     }
 
-    fn set(&mut self, _key: StorageKey, _value: StorageValue) {
-        todo!()
+    fn set(&mut self, key: StorageKey, value: StorageValue) {
+        let cache_key = key.into();
+        let cache_value = value.into();
```

Comment:
non-blocking: It's not clear at a glance how costly these conversions are. Have we considered
1. Switching to `as_`, which is zero-cost by convention or
2. Replacing the `CacheKey` and `CacheValue` types with `StorageKey` and `StorageValue`

Reply:
Added `as_` methods.